File: c++/Contains_Duplicate.cpp
/*Given an array of integers, find if the array contains any duplicates. Your function should return true if any value appears at least twice in the array, and it should return false if every element is distinct.*/
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        map<int,bool> mp;
        for(int i = 0; i < nums.size(); i++){
            if(mp.find(nums[i]) != mp.end()){
               return true;
            }
            else
            {
                mp[nums[i]] = true;
            }
        }
        return false;
    }
};

File: python/Contains_Duplicate.py
"""Given an array of integers, find if the array contains any duplicates. Your function should return true if any value appears at least twice in the array, and it should return false if every element is distinct."""
class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        mp = {}
        for num in nums:
            if mp.has_key(num): return True
            mp[num] = True
        return False    
